### PR TITLE
Feature simple page pagination

### PR DESCRIPTION
**Pull request by maZahaca**
* Added pagination through the links
* Added Pagination.maxResultsCount option and fixed maxPagesCount option
* Added Pagination.timeout and Action.waitForPageTimeout options.

#17

### CODE REVIEW

**Comment by maZahaca** [comment]
@jifeon for review

File: lib/Paginator.js [hunk 1]
```diff
@@ -131,11 +152,32 @@ Paginator.prototype = {
...
+                            debug('Set paginateStrategy to %s', this._paginateStrategy);
+                        }, this),
+                        this._hasPageDataChanged(previousPageHtml).then(function () {
+                            this._paginateStrategy = this.STRATEGY.AJAX;
+                            debug('Set paginateStrategy to %s', this._paginateStrategy);
+                        }, this)
+                    ]);
+                }
+                else {
+                    switch (this._paginateStrategy) {
```
Comment (jifeon) [inline lib/Paginator.js]: why `if-else` + `switch` instead of only `switch` with `case null`?

File: lib/Paginator.js [hunk 2]
```diff
@@ -151,11 +193,37 @@ Paginator.prototype = {
...
+     */
+    _hasPageScrolled: function (previousPageHeight) {
+        return this._actions.wait(function () {
+            return document.body.scrollHeight;
+        }, function (scrollHeight) {
+            return scrollHeight > previousPageHeight;
+        }, [], this._timeout);
+    },
+
+    _hasPageDataChanged: function (previousPageHtml) {
```
Comment (jifeon) [inline lib/Paginator.js]: jsdoc

File: lib/Parser.js [hunk 3]
```diff
@@ -23,6 +23,8 @@ var debugLib = require('debug'),
  * @property {string} type
  * @property {?string} scope
  * @property {?string} parentScope
+ * @property {?boolean} waitForPage
+ * @property {?number} waitForPageTimeout
```
Comment (jifeon) [inline lib/Parser.js]: where is documentation changes?

File: lib/Parser.js [hunk 4]
```diff
@@ -411,6 +413,13 @@ Parser.prototype = {
                     .then(function (pageResults) {
                         debug('Pagination results %o', pageResults);
                         Array.prototype.push.apply(results, pageResults);
+                        if (
+                            this._paginator.getMaxResultsCount() !== Infinity &&
```
Comment (jifeon) [inline lib/Parser.js]: why you need first condition?

File: lib/Parser.js [hunk 5]
```diff
@@ -411,6 +413,13 @@ Parser.prototype = {
...
+                            results.length > this._paginator.getMaxResultsCount() - 1
+                        ) {
+                            results = results.slice(0, this._paginator.getMaxResultsCount() - 1);
```
Comment (jifeon) [inline lib/Parser.js]: save `this._paginator.getMaxResultsCount() - 1` to var

File: lib/Parser.js (hunk 3 above)
Comment (maZahaca) [inline lib/Parser.js]: added

**Pull request merged**